Automattic/wp-calypso
# BBE: Add input to accept search terms

Fixes Automattic/martech#1556

## Proposed Changes

* Adds a new text field to the site options step in the BBE flow.
<img width="1920" alt="image" src="https://user-images.githubusercontent.com/5436027/220844846-adbde266-8e07-4a84-9165-ecfcfdeb05a5.png">

* On the content collection form, modified the copy for the Home page:
<img width="921" alt="image" src="https://user-images.githubusercontent.com/5436027/220844938-df8fd2ca-a8d7-4f9b-a352-81e24e2d7209.png">
 

## Testing Instructions



* Apply D102526-code.
* Go to `/start/do-it-for-me`.
* Select any of the options on the New/Existing site screen.
* On the site options screen, confirm that you can see the new text field for accepting search terms.
* Continue through the flow and complete the purchase.
* On the content collection form, confirm that the copy for the Home page matches the one in the screenshot above.
* Additional testing instructions in D102526-code.

## Pre-merge Checklist



- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)? (TODO)
